pkg/validation/internal/good_practices.go

```diff
@@ -116,6 +130,68 @@ func validateHubChannels(channels []string) error {
 	return nil
 }
 
+// validateRBACForCRDsWith to warning when/if permissions to create CRD are found in the rules
+func validateRBACForCRDsWith(csv *operatorsv1alpha1.ClusterServiceVersion) error {
+	apiGroupResourceMap := map[string][]string{
+		"apiextensions.k8s.io": {"customresourcedefinitions", "*", "[*]"},
+	}
+	verbs := []string{"create", "*", "[*]", "patch"}
+	warning := goerrors.New("CSV contains permissions to create CRD. An Operator shouldn't deploy or manage " +
+		"other operators (such patterns are known as meta or super operators or include CRDs in its Operands)." +
+		" It's the Operator Lifecycle Manager's job to manage the deployment and lifecycle of operators. " +
+		" Please, review the design of your solution and if you should not be using Dependency Resolution from OLM instead")
+
+	for _, perm := range csv.Spec.InstallStrategy.StrategySpec.Permissions {
+		if hasRBACFor(perm, apiGroupResourceMap, verbs) {
+			return warning
+		}
+	}
+
+	for _, perm := range csv.Spec.InstallStrategy.StrategySpec.ClusterPermissions {
+		if hasRBACFor(perm, apiGroupResourceMap, verbs) {
+			return warning
+		}
+	}
+
+	return nil
+}
+
+func hasRBACFor(perm v1alpha1.StrategyDeploymentPermissions, apiGroupResourceMap map[string][]string, verbs []string) bool {
+	if perm.Rules == nil {
+		return false
+	}
+
+	// For each APIGroup and list of resources that we are looking for
+	for apiFromMap, resourcesFromMap := range apiGroupResourceMap {
+		for _, rule := range perm.Rules {
+			for _, api := range rule.APIGroups {
+				// If we found the APIGroup
+				if api == apiFromMap {
+					for _, res := range rule.Resources {
+						for _, resFromMap := range resourcesFromMap {
+							// If we found the resource
+							if resFromMap == res {
+								// Check if we find the verbs:
+								for _, verbFromList := range verbs {
+									for _, ruleVerb := range rule.Verbs {
+										// If we found the verb
+										if verbFromList == ruleVerb {
+											// stopping by returning true
+											return true
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
I think that we can simplify this using `k8s.io/apimachinery/pkg/util/sets`:

```suggestion
    verbSet := sets.NewString(verbs...)
    for _, rule := range perm.Rules {
    	for group, resources := range apiGroupResourcesMap {
    	    if sets.NewString(rule.APIGroups).Has(group) && 
    	      sets.NewString(resources).Has(rule.Resource) 
    	      verbSet.HasAny(rule.Verbs...) {
    	        return true
            }
        }
    }
```

Reply:
I didn't fully test it so it probably needs to be tweaked a bit.

I think using sets for quick lookups is easier to read than a bunch of nested loops, but its your call. Its a suggestion, not requested changes.

Reply:
I tried to check it, and it did not work well as expected.
Also, I found little hard to understand the code with the proposed changes.
So, do you think that we can move forward without applying this one?
If yes, could you lgtm this PR?